### PR TITLE
Decrement active_appenders metric when no samples added

### PR DESCRIPTION
**Pull request by bboreham**
This fixes an edge case that I noticed in our metrics, where `tsdb_head_active_appenders` was increasing steadily all day.

Also add a test that the metric is incremented and decremented as expected with and without samples.

### CODE REVIEW

File: tsdb/head_test.go [hunk 1]
```diff
@@ -402,6 +402,39 @@ func TestHead_WALMultiRef(t *testing.T) {
...
+
+	// First rollback with no samples
+	app := head.Appender(context.Background())
+	require.Equal(t, 1.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+	require.NoError(t, app.Rollback())
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+
+	// Then commit with no samples
+	app = head.Appender(context.Background())
+	require.NoError(t, app.Commit())
```
Comment (codesome) [inline tsdb/head_test.go:419]: Like for rollback, we should check if it was 1.0 just before this line

File: tsdb/head_test.go [hunk 2]
```diff
@@ -402,6 +402,39 @@ func TestHead_WALMultiRef(t *testing.T) {
...
+	require.NoError(t, err)
+	require.Equal(t, 1.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+	require.NoError(t, app.Rollback())
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+
+	// Now rollback with one sample
+	app = head.Appender(context.Background())
+	_, err = app.Append(0, labels.FromStrings("foo", "bar"), 100, 1)
+	require.NoError(t, err)
+	require.NoError(t, app.Commit())
```
Comment (codesome) [inline tsdb/head_test.go:434]: Like for rollback, we should check if it was 1.0 just before this line

File: tsdb/head_test.go [hunk 3]
```diff
@@ -402,6 +402,39 @@ func TestHead_WALMultiRef(t *testing.T) {
...
+
+	// Now rollback with one sample
+	app = head.Appender(context.Background())
+	_, err := app.Append(0, labels.FromStrings("foo", "bar"), 100, 1)
+	require.NoError(t, err)
+	require.Equal(t, 1.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+	require.NoError(t, app.Rollback())
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+
+	// Now rollback with one sample
```
Comment (codesome) [inline tsdb/head_test.go:430]:
```suggestion
	// Now commit with one sample.
```

And full stop at the end of every comment in this file :)

**Review by codesome — COMMENTED** [review]
LGTM, some nits

File: tsdb/head_test.go (hunk 1 above)
Comment (bboreham) [reply tsdb/head_test.go:419]: Under what model could the behaviour vary between the two cases?

**Review by codesome — APPROVED** [review]

**Pull request merged**